checkmarx-ltd/cx-common-js-client
# Plug 1446

**Changes** 
- When login with SSO -> used API connect/userinfo to fetch permissions
- when login with uname_password - using jwt decode to fetch permissions
- Removed hardcode values of result states
- Fetched result state dynamically and shown according to permission

Test below cases login with SSO and username_password both
- Create a new Role R1
- Create a new user with R1 roles

**Test Case 1**
- Role R1 - uncheck permission of set-result-state-toverify.
- login with the VS code plugin and check the result state available in the result state table
Expected output - To Verify option not available in Result state dropdown
- Do similar things for Confirmed, Urgent, Not exploitable, Proposed not exploitable
- update vulnerability states of any
- Check with different language 


**Test Case 2**
- Create a custom state as per documentation
https://checkmarx.atlassian.net/wiki/spaces/KC/pages/5982847752/Adding+Custom+Result+States
- Provide newly created states permission to the user 
- log in with the VS plugin and check the result state available in the result state table
Expected output - newly created custom field display in result state dropdown
- update the vulnerability states of any
- remove permission of newly created custom states
- login with the VS plugin and check result state is available in the result state table or not
Expected output - option not available in Result state dropdown



**Test Case 3**
- Test all this with latest SAST 9.4,9.5,9.6.

**Test Case 4**
- For SAST 9.4 API is not available - sast/resultStates
- If user not have permissions and try to update selected vulnarabilty result state then it will return error message -You are not authorized to mark the selected vulnerability result as
